### DISCUSSION TITLE
PatchInferer and Monai server

### DISCUSSION THREAD

**Discussion by mdrio** [Project-MONAI/MONAILabel#1594] [Q&A]
Hi, I am trying to do some inference with a TorchScript model on whole slide images using a MONAILabelApp and DSA. I am able to use the model using the PatchInferer and the WSISlidingWindowSplitter. How can it be integrated in a MONAILabelApp? I checked the sample-apps/pathology, but apparentely only SimpleInferer is used. 
Thanks

**Comment by MathijsdeBoer** [comment]
Perhaps you could make use of the `monai.inferers.SlidingWindowInferer`? I'm not experienced with whole slide images, so I'm not sure whether that'll work for you.

**Reply by MathijsdeBoer** [reply]
Rereading I realized that you already know how to run a prediction. I think you can just change the `InferTask` class in ".../lib/infers/" to use the `PatchInferer` instead. For example, `apps/radiology` has a few examples that define the `inferer()` method as follows:

```python
    def inferer(self, data=None) -> Inferer:
        return SlidingWindowInferer(
            roi_size=self.roi_size,
            sw_batch_size=2,
            overlap=0.5,
            padding_mode="replicate",
            mode="gaussian",
        )
```

**Reply by mdrio** [reply]
Ok, thanks. Actually I am using PatchInferer and SlidingWindowSplitter. Anyway I got it working, but now I have the following problem: since the model is patch based , the output is rescaled. So if a 2560x2560 region is processed with 256x256 patches, the output is 10x10. I correctly get the annotation (segmentation) in DSA format using FindContoursd transform (present in MonaiLabel app code), but rescaled. Is there any utility I can use for getting the result with the right scale? O do I have to do it manually (perhaps with Shapely)?

**Reply by MathijsdeBoer** [reply]
Perhaps `monailabel.transform.post.Restored` can help you out? The radiology apps seem to use it to return an output to match the input(s).
